plugins/chat/lib/chat_sdk/message.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+module ChatSDK
+  class Message
+    include Chat::WithServiceHelper
+
+    # Creates a new message in a chat channel.
+    #
+    # @param raw [String] The content of the message.
+    # @param channel_id [Integer] The ID of the chat channel.
+    # @param guardian [Guardian] The user's guardian object, for policy enforcement.
+    # @param in_reply_to_id [Integer, nil] The ID of the message this is in reply to (optional).
+    # @param thread_id [Integer, nil] The ID of the thread this message belongs to (optional).
+    # @param upload_ids [Array<Integer>, nil] The IDs of any uploads associated with the message (optional).
+    # @param streaming [Boolean] Whether the message is part of a streaming operation (default: false).
+    # @yield [helper, message] Offers a block with a helper and the message for streaming operations.
+    # @yieldparam helper [Helper] The helper object for streaming operations.
+    # @yieldparam message [Message] The newly created message object.
+    # @return [ChMessage] The created message object.
+    #
+    # @example Creating a simple message
+    #   ChatSDK::Message.create(raw: "Hello, world!", channel_id: 1, guardian: Guardian.new)
+    #
+    # @example Creating a message with a block for streaming
+    #   Message.create_with_stream(raw: "Streaming message", channel_id: 1, guardian: Guardian.new) do |helper, message|
+    #     helper.stream(raw: "Continuation of the message")
+    #   end
+    def self.create(**params, &block)
+      new.create(**params, &block)
+    end
+
+    # Creates a new message with streaming enabled by default.
+    #
+    # This method is a convenience wrapper around `create` with `streaming: true` set by default.
+    # It supports all the same parameters and block usage as `create`.
+    #
+    # @see #create
+    def self.create_with_stream(**params, &block)
+      self.create(**params, streaming: true, &block)
+    end
+
+    def create(
+      raw:,
+      channel_id:,
+      guardian:,
+      in_reply_to_id: nil,
+      thread_id: nil,
+      upload_ids: nil,
+      streaming: false,
+      &block
+    )
+      message =
+        with_service(
+          Chat::CreateMessage,
+          message: raw,
+          guardian: guardian,
+          chat_channel_id: channel_id,
+          in_reply_to_id: in_reply_to_id,
+          thread_id: thread_id,
+          upload_ids: upload_ids,
+          streaming: streaming,
+        ) do
+          on_model_not_found(:channel) { raise "Couldn't find channel with id: `#{channel_id}`" }
```

Comment:
Not sure about this but should we support multiple languages on the SDK errors? Should we have a translation file for the SDK?

Reply:
Yes I hesitated, I think that's ok for now. But I wîll probably rethink this.